File: codegen/objc/objc-entities/src/main/groovy/com/stanfy/helium/handler/codegen/objectivec/entity/ObjCFile.java
package com.stanfy.helium.handler.codegen.objectivec.entity;
import com.stanfy.helium.handler.codegen.objectivec.entity.file.ObjCSourcePart;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by ptaykalo on 8/17/14.
 * Structure that represent Objective-C source file
 * Concrete implementations can be header, and implementation file
 */
public abstract class ObjCFile implements ObjCSourcePart {

  private String name;
  private ArrayList<ObjCSourcePart> sourceParts = new ArrayList<ObjCSourcePart>();

  public ObjCFile(final String name) {
    this.name = name;
  }

  /**
   * Returns file name
   */
  public String getName() {
    return name;
  }

  /**
   * File extension :)
   */
  public abstract String getExtension();


  public List<ObjCSourcePart> getSourceParts() {
    return sourceParts;
  }

  public void addSourcePart(final ObjCSourcePart sourcePart) {
    sourceParts.add(sourcePart);
  }

  @Override
  public String asString() {
    StringBuilder bld = new StringBuilder();
    for (ObjCSourcePart part : sourceParts) {
      bld.append(part.asString());
    }
    return bld.toString();
  }
}

File: codegen/objc/objc-entities/src/main/kotlin/com/stanfy/helium/handler/codegen/objectivec/entity/builder/ObjCFile.kt
package com.stanfy.helium.handler.codegen.objectivec.entity.builder

import com.stanfy.helium.handler.codegen.objectivec.entity.file.ObjCSourcePart

import java.util.ArrayList
import java.util.List

/**
 * Created by ptaykalo on 8/17/14.
 * Structure that represent Objective-C source file
 * Concrete implementations can be header, and implementation file
 */
public abstract class ObjCFile(val name: String) : ObjCSourcePart {

  public val sourceParts = ArrayList<ObjCSourcePart>()

  /**
   * File extension :)
   */
  public abstract fun getExtension(): String


  public fun addSourcePart(sourcePart: ObjCSourcePart) {
    sourceParts.add(sourcePart)
  }

  override fun asString(): String {
    val bld = StringBuilder()
    for (part in sourceParts) {
      bld.append(part.asString())
    }
    return bld.toString()
  }
}
